checkov/common/graph/graph_builder/variable_rendering/renderer.py

```diff
@@ -0,0 +1,100 @@
+        loops = 0
+        while len(edges_to_render) > 0:
+            logging.info(f"evaluating {len(edges_to_render)} edges")
+            # group edges that have the same origin and label together
+            edges_groups = self.group_edges_by_origin_and_label(edges_to_render)
+            if self.run_async:
+                run_function_multithreaded(
+                    func=self._edge_evaluation_task,
+                    data=edges_groups,
+                    max_group_size=1,
+                    num_of_workers=self.max_workers,
+                )
+            else:
+                for edge_group in edges_groups:
+                    self._edge_evaluation_task([edge_group])
+            for edge in edges_to_render:
+                origin = edge.origin
+                if origin not in self.done_edges_by_origin_vertex:
+                    self.done_edges_by_origin_vertex[origin] = []
+                self.done_edges_by_origin_vertex[origin].append(edge)
+
+            for edge in edges_to_render:
+                origin_vertex_index = edge.origin
+                out_edges = self.local_graph.out_edges.get(origin_vertex_index, [])
+                if all(e in self.done_edges_by_origin_vertex.get(origin_vertex_index, []) for e in out_edges):
+                    end_vertices_indexes.append(origin_vertex_index)
+            edges_to_render = self.local_graph.get_in_edges(end_vertices_indexes)
+            edges_to_render = list(
+                set(
+                    [
+                        edge
+                        for edge in edges_to_render
+                        if edge not in self.done_edges_by_origin_vertex.get(edge.origin, [])
+                    ]
+                )
+            )
+            loops += 1
+            if loops >= self.MAX_NUMBER_OF_LOOPS:
+                logging.warning(f"Reached 50 graph edge iterations, breaking.")
+                break
+
+        self.local_graph.update_vertices_configs()
+        logging.info("done evaluating edges")
+        self.evaluate_non_rendered_values()
+        logging.info("done evaluate_non_rendered_values")
+
+    def _edge_evaluation_task(self, edges: List[List[Edge]]) -> List[Edge]:
+        inner_edges = edges[0]
+        self.evaluate_vertex_attribute_from_edge(inner_edges)
+        return inner_edges
+
+    @abstractmethod
+    def evaluate_vertex_attribute_from_edge(self, edge_list: List[Edge]) -> None:
+        pass
+
+    @staticmethod
+    def group_edges_by_origin_and_label(edges: List[Edge]) -> List[List[Edge]]:
+        edge_groups: Dict[str, List[Edge]] = {}
+        for edge in edges:
+            origin_and_label_hash = calculate_hash(f"{edge.origin}{edge.label}")
+            if not edge_groups.get(origin_and_label_hash):
+                edge_groups[origin_and_label_hash] = []
+            edge_groups[origin_and_label_hash].append(edge)
+        return list(edge_groups.values())
```

Comment:
```suggestion
        return list(edge_groups)
```

Reply:
`list(edge_groups)` is syntactically equal to `list(edge_groups.keys())`